pkg/types/service_plan.go

```diff
@@ -59,6 +59,11 @@ func (e *ServicePlan) Validate() error {
 	if e.ServiceOfferingID == "" {
 		return fmt.Errorf("service plan service offering id missing")
 	}
+	if cliFriendly, err := util.IsCLIFriendly(e.CatalogName); err != nil {
+		return err
```

Comment:
this is actually an internal error and should not be returned. This will return internal error as 400. Maybe log the actual error and return something generic such as `Could not validate ___ name due to internal error. Check logs for more information.`

Reply:
its **not** an internal error - a user registers his broker and this user knows his broker credentials so this user can access the catalog in the first place - the broker as well as the broker catalog are user INPUT. Disclosing information about the catalog does not disclose anything internal as the user can access the catalog in the first place.

If this broker's catalog has a not-cli-friendly entity name we should let the user know exactly whats wrong with his broker catalog and not make him open a ticket in order to figure that out.

Reply:
The error is actually not being able to compile the regex. If the name is valid or not is returned as a boolean. That’s why it’s internal error.